src/components/accession2/withdraw/CountWithdrawal.tsx

```diff
@@ -0,0 +1,155 @@
+import { Checkbox, Textfield } from '@terraware/web-components';
+import { Accession, Withdrawal } from 'src/types/Accession';
+
+export interface CountWithdrawalProps {
+  accession: Accession;
+  onWithdrawCtUpdate: (withdrawnQuantity: Withdrawal['withdrawnQuantity'], valid: boolean) => void;
+}
+
+export default function CountWithdrawal(props: CountWithdrawalProps): JSX.Element {
+  const { accession, onWithdrawCtUpdate } = props;
+
+  const [withdrawAllSelected, setWithdrawAllSelected] = useState(false);
+  const [withdrawnQuantity, setWithdrawnQuantity] = useState<Withdrawal['withdrawnQuantity']>({
+    quantity: 0,
+    units: 'Seeds',
+  });
+  const [withdrawnQtyError, setWithdrawnQtyError] = useState<string>('');
+
+  useEffect(() => {
+    setWithdrawnQtyError('');
+    if (!accession.estimatedCount) {
+      setWithdrawnQtyError(strings.MISSING_SUBSET_WEIGHT_ERROR_COUNT);
+    }
+  }, [accession.estimatedCount]);
+
+  const onChangeWithdrawnQuantity = (value: number) => {
+    let valid = false;
+
+    if (
+      value.toString() === accession.remainingQuantity?.quantity.toString() &&
+      withdrawnQuantity?.units === accession.remainingQuantity?.units
+    ) {
+      setWithdrawAllSelected(true);
+    } else {
+      setWithdrawAllSelected(false);
+    }
+
+    setWithdrawnQuantity(
+      value.toString().trim() === ''
+        ? undefined
+        : {
+            quantity: value || 0,
+            units: 'Seeds',
+          }
+    );
+
+    setWithdrawnQtyError('');
+    if (!accession.estimatedCount) {
+      setWithdrawnQtyError(strings.MISSING_SUBSET_WEIGHT_ERROR_COUNT);
+    }
+
+    valid = validateAmount(value);
+    onWithdrawCtUpdate(
+      {
+        quantity: value || 0,
+        units: 'Seeds',
+      },
+      valid
+    );
+  };
+
+  const onSelectAll = (id: string, withdrawAll: boolean) => {
+    if (withdrawAll) {
+      if (
+        accession.remainingQuantity?.units &&
+        accession.remainingQuantity?.units !== 'Seeds' &&
+        accession.estimatedCount
+      ) {
+        onChangeWithdrawnQuantity(accession.estimatedCount);
+      } else if (accession.remainingQuantity?.units && accession.remainingQuantity?.units === 'Seeds') {
+        onChangeWithdrawnQuantity(accession.remainingQuantity?.quantity);
+      }
+    } else {
+      setWithdrawnQuantity(undefined);
+    }
+
+    setWithdrawAllSelected(withdrawAll);
+  };
+
+  const validateAmount = (withdrawnQty: number) => {
+    if (withdrawnQty) {
+      if (!accession.estimatedCount) {
+        setWithdrawnQtyError(strings.MISSING_SUBSET_WEIGHT_ERROR_COUNT);
+        return false;
+      }
+      if (isNaN(withdrawnQty)) {
+        setWithdrawnQtyError(strings.INVALID_VALUE);
+        return false;
+      }
+      if (Number(withdrawnQty) <= 0) {
+        setWithdrawnQtyError(strings.INVALID_VALUE);
+        return false;
+      }
+      if (accession.remainingQuantity?.units && Number(accession.remainingQuantity.units === 'Seeds')) {
```

Comment:
```suggestion
      if (accession.remainingQuantity?.units === 'Seeds') {
```

Reply:
Need to keep the check for units because it can be undefined.